File: ebean-querybean/src/test/java/org/example/domain/otherpackage/ValidEmailAttributeConverter.java
package org.example.domain.otherpackage;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;

@Converter
public class ValidEmailAttributeConverter implements AttributeConverter<ValidEmail, String> {
  @Override
  public String convertToDatabaseColumn(final ValidEmail attribute) {
    return attribute.getEmailAddress();
  }

  @Override
  public ValidEmail convertToEntityAttribute(final String dbData) {
    return new ValidEmail(dbData);
  }
}

File: tests/test-kotlin/src/main/kotlin/org/example/otherpackage/ValidEmailAttributeConverter.kt
package org.example.otherpackage

import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter

@Converter
class ValidEmailAttributeConverter : AttributeConverter<ValidEmail, String> {
  override fun convertToDatabaseColumn(attribute: ValidEmail): String {
    return attribute.emailAddress
  }

  override fun convertToEntityAttribute(dbData: String): ValidEmail {
    return ValidEmail(dbData)
  }
}
